src/libs/antares/study/include/antares/study/parts/thermal/cluster.h

```diff
@@ -307,6 +317,7 @@ class ThermalCluster final: public Cluster, public std::enable_shared_from_this<
     CostProvider& getCostProvider();
 
 private:
+    bool anyMinGen_ = false;
```

Comment:
Not necessary

Reply:
Replaced with `std::optional<bool> hasMinGen_;`, as requested